File: app/ios/RunninWatch Watch App/Screens/PreRunScreen.swift
import SwiftUI

/// Jornada 1, Passo 1/5 — TIPO de corrida. Espelha o Step 0 do prep_page do
/// iPhone (TypeStep). User pode abrir Runnin direto no Watch e escolher:
///   - "SESSÃO DO DIA" (se houver plano com sessão pra hoje, vinda do iPhone)
///   - "CORRIDA LIVRE" (sempre disponível)
///
/// Tap NÃO inicia a corrida — apenas avança pra BriefingScreen (Passo 5/5).
/// Esse desacoplamento garante que user lê o briefing antes de confirmar.
struct PreRunScreen: View {
    @EnvironmentObject var state: WatchRunState

    var body: some View {
        ScrollView {
            VStack(alignment: .center, spacing: 10) {
                HStack {
                    RunninLogo()
                    Spacer()
                    Text("1/5")
                        .font(.system(size: 9, weight: .medium, design: .monospaced))
                        .tracking(0.8)
                        .foregroundStyle(.white.opacity(0.45))
                }
                .padding(.top, 4)

                Text("ESCOLHA O TIPO")
                    .font(.system(size: 9, weight: .medium, design: .monospaced))
                    .tracking(1.5)
                    .foregroundStyle(.white.opacity(0.5))
                    .padding(.top, 4)

                if let session = state.todaySession {
                    runButton(
                        title: "SESSÃO DO DIA",
                        subtitle: "\(session.type) · \(formattedKm(session.distanceKm))",
                        accent: true
                    ) {
                        state.localStep = .briefing(SelectedRunType(
                            type: session.type,
                            planSessionId: session.planSessionId,
                            distanceKm: session.distanceKm
                        ))
                    }
                }

                runButton(
                    title: "CORRIDA LIVRE",
                    subtitle: "Sem plano · sem alvo",
                    accent: false
                ) {
                    state.localStep = .briefing(SelectedRunType(
                        type: "Free Run",
                        planSessionId: nil,
                        distanceKm: nil
                    ))
                }
            }
            .padding(.horizontal, 4)
        }
    }

    @ViewBuilder
    private func runButton(
        title: String, subtitle: String, accent: Bool,
        action: @escaping () -> Void
    ) -> some View {
        Button(action: action) {
            VStack(alignment: .leading, spacing: 4) {
                Text(title)
                    .font(.system(size: 12, weight: .heavy, design: .monospaced))
                    .tracking(1.0)
                    .foregroundStyle(accent ? Color.black : Color.white)
                Text(subtitle)
                    .font(.system(size: 9, weight: .medium, design: .monospaced))
                    .foregroundStyle(accent ? Color.black.opacity(0.7) : Color.white.opacity(0.5))
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(.vertical, 8)
            .padding(.horizontal, 10)
            .background(accent ? Color.cyan : Color.white.opacity(0.08))
            .overlay(
                RoundedRectangle(cornerRadius: 6)
                    .stroke(accent ? Color.clear : Color.cyan.opacity(0.3), lineWidth: 1)
            )
            .clipShape(RoundedRectangle(cornerRadius: 6))
        }
        .buttonStyle(.plain)
    }

    private func formattedKm(_ km: Double) -> String {
        if km == km.rounded() { return "\(Int(km))km" }
        return String(format: "%.1fkm", km)
    }
}

File: app/android/wear/src/main/kotlin/com/s6lab/runnin/wear/screens/PreRunScreen.kt
package com.s6lab.runnin.wear.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.wear.compose.material.Text
import com.s6lab.runnin.wear.components.RunninLogo
import com.s6lab.runnin.wear.models.LocalStep
import com.s6lab.runnin.wear.models.SelectedRunType
import com.s6lab.runnin.wear.models.WatchRunState
import com.s6lab.runnin.wear.theme.scaledFont

/**
 * Jornada 1, Passo 1/5 — TIPO de corrida. Espelha o Step 0 do prep_page do
 * phone (TypeStep).
 *
 * Paralelo do `PreRunScreen` do Apple Watch (Screens/PreRunScreen.swift).
 */
@Composable
fun PreRunScreen() {
    val state = WatchRunState.shared
    val scroll = rememberScrollState()
    val accent = state.accentColor

    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(scroll)
            .padding(horizontal = 4.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        // Logo centralizada — em display redondo top-LEFT é cortado.
        androidx.compose.foundation.layout.Box(
            modifier = Modifier.fillMaxWidth().padding(top = 4.dp),
            contentAlignment = androidx.compose.ui.Alignment.Center,
        ) {
            RunninLogo()
        }

        Text(
            text = "ESCOLHA O TIPO",
            color = Color.White.copy(alpha = 0.5f),
            style = scaledFont(9f, FontWeight.Medium),
            modifier = Modifier.padding(top = 4.dp),
        )

        val today = state.todaySession
        if (today != null) {
            if (today.isExecuted) {
                CompletedSessionCard(typeLabel = today.type, distanceKm = today.distanceKm)
            } else {
                RunButton(
                    title = "SESSÃO DO DIA",
                    subtitle = "${today.type} · ${formattedKm(today.distanceKm)}",
                    accent = true,
                    accentColor = accent,
                    onClick = {
                        state.localStep = LocalStep.Briefing(
                            SelectedRunType(today.type, today.planSessionId, today.distanceKm)
                        )
                    },
                )
            }
        }

        RunButton(
            title = "CORRIDA LIVRE",
            subtitle = "Sem plano · sem alvo",
            // Quando a sessão do dia já foi feita, Free Run vira a opção
            // principal (accent) — espelha o default do phone.
            accent = today?.isExecuted == true,
            accentColor = accent,
            onClick = {
                state.localStep = LocalStep.Briefing(
                    SelectedRunType("Free Run", null, null)
                )
            },
        )
    }
}

@Composable
private fun RunButton(
    title: String,
    subtitle: String,
    accent: Boolean,
    accentColor: Color,
    onClick: () -> Unit,
) {
    val bg = if (accent) accentColor else Color.White.copy(alpha = 0.08f)
    val borderColor = if (accent) Color.Transparent else accentColor.copy(alpha = 0.3f)
    val titleColor = if (accent) Color.Black else Color.White
    val subtitleColor = if (accent) Color.Black.copy(alpha = 0.7f) else Color.White.copy(alpha = 0.5f)
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(6.dp))
            .background(bg)
            .border(1.dp, borderColor, RoundedCornerShape(6.dp))
            .clickable(onClick = onClick)
            .padding(horizontal = 10.dp, vertical = 8.dp),
    ) {
        Text(title, color = titleColor, style = scaledFont(12f, FontWeight.Black))
        Spacer(Modifier.size(4.dp))
        Text(subtitle, color = subtitleColor, style = scaledFont(9f, FontWeight.Medium))
    }
}

@Composable
private fun CompletedSessionCard(typeLabel: String, distanceKm: Double) {
    val green = Color(0xFF36D399)
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(6.dp))
            .background(green.copy(alpha = 0.08f))
            .border(1.dp, green.copy(alpha = 0.45f), RoundedCornerShape(6.dp))
            .padding(horizontal = 10.dp, vertical = 8.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Text("✓", color = green, style = scaledFont(11f, FontWeight.Black))
            Spacer(Modifier.width(6.dp))
            Text("SESSÃO CONCLUÍDA", color = green, style = scaledFont(10f, FontWeight.Black))
        }
        Text(
            "$typeLabel · ${formattedKm(distanceKm)}",
            color = Color.White.copy(alpha = 0.55f),
            style = scaledFont(9f, FontWeight.Medium),
        )
    }
}

private fun formattedKm(km: Double): String =
    if (km == km.toInt().toDouble()) "${km.toInt()}km"
    else "%.1fkm".format(km)
